api/src/migrations/geoprocessing/1611221157285-initialGeoDBSetup.ts

```diff
@@ -0,0 +1,185 @@
+        "type" shape_type,
+        "size" int
+      ) PARTITION BY LIST (type);
+
+      CREATE TABLE planing_units_geom_square PARTITION OF planing_units_geom FOR VALUES IN ('square');
+      CREATE TABLE planing_units_geom_hexagon PARTITION OF planing_units_geom FOR VALUES IN ('hexagon');
+      CREATE TABLE planing_units_geom_irregular PARTITION OF planing_units_geom FOR VALUES IN ('irregular');
+
+      -- Planning units usage by scenario.
+      CREATE TABLE "scenarios_pu_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "pu_geomid" uuid NOT NULL REFERENCES "planing_units_geom" ("id"),
+        "scenario_id" uuid NOT NULL,
+        "puid" int NOT NULL,
+        "lockin_status" int,
+        "xloc" float8,
+        "yloc" float8,
+        "protected_area" float8
+      );
+
+      -- Cost data for planning units usage by scenario.
+      CREATE TABLE "scenarios_pu_cost_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "output_results_data_id" uuid NOT NULL REFERENCES "scenarios_pu_data" ("id"),
+        "cost" float8
+      );
+
+      CREATE TABLE "output_results_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "puid" int REFERENCES "scenarios_pu_data" ("puid"),
+        "scenario_id" uuid,
+        "run_id" uuid,
+        "value" float8,
+        "missing_values" jsonb
+      );
+
+
+      COMMENT ON TABLE "admin_regions" IS 'will cover adm0,adm1 and adm2';
+      COMMENT ON TABLE "features_data" IS 'For geometry storage';
+      COMMENT ON COLUMN "features_data"."properties" IS 'Either have the properties as a jsonb or as columns on this tables';
+      COMMENT ON COLUMN "planing_units_geom"."size" IS 'represent km';
+      COMMENT ON TABLE "scenarios_pu_data" IS 'marxan view from an output_results';
+      COMMENT ON TABLE "scenarios_pu_cost_data" IS 'marxan view from an output_results';
+      COMMENT ON TABLE "output_results_data" IS 'output_results from marxan runs under an scenario';
+
+      `);
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {
+      await queryRunner.query(`
+      DROP TABLE IF EXISTS "output_results_metadata";
+      DROP TABLE IF EXISTS "output_results_data";
+      DROP TABLE IF EXISTS "scenarios_pu_cost_data";
+      DROP TABLE IF EXISTS "scenarios_pu_data";
+      DROP TABLE IF EXISTS "scenarios_metadata";
+      DROP TABLE IF EXISTS "planing_units_geom";
+      DROP TABLE IF EXISTS "scenario_features_data";
+      DROP TABLE IF EXISTS "features_data";
+      DROP TABLE IF EXISTS "admin_regions";
+      DROP TABLE IF EXISTS "wdpa";
+
+
+      DROP TYPE IF EXISTS source_type;
+      DROP TYPE IF EXISTS shape_type;
+      DROP TYPE IF EXISTS status;
+
+      DROP EXTENSION postgis_topology; -- OPTIONAL
+      DROP EXTENSION postgis_raster; -- OPTIONAL
+      DROP EXTENSION postgis;
+      DROP EXTENSION plpgsql;
+      DROP EXTENSION tablefunc;
+      DROP EXTENSION uuid-ossp;
```

Comment:
should be wrapped in guard for pg < 13; personally I would not bother about dropping `uuid-ossp` and `plpgsql`, same as `pgcrypto` in the migration where I added this - in theory not even the postgis stuff as we'd need it for sure and one it's enabled it's enabled - but ok to drop everything, ofc 😄